File: mobile_app/lib/models/property_brief.dart
class PropertyBrief {
  final String id;
  final String propertyType;
  final String city;
  final int sizeSqft;
  final int budgetMin;
  final int budgetMax;
  final String timeline;
  final String scope;
  final String status;
  final String createdAt;
  final int quoteCount;
  final List<String> imageUrls;
  final String? customerName;

  PropertyBrief({
    required this.id,
    required this.propertyType,
    required this.city,
    required this.sizeSqft,
    required this.budgetMin,
    required this.budgetMax,
    required this.timeline,
    required this.scope,
    required this.status,
    required this.createdAt,
    required this.quoteCount,
    required this.imageUrls,
    this.customerName,
  });

  factory PropertyBrief.fromJson(Map<String, dynamic> json) {
    return PropertyBrief(
      id: json['id']?.toString() ?? '',
      propertyType: json['propertyType']?.toString() ?? '',
      city: json['city']?.toString() ?? '',
      sizeSqft: (json['sizeSqft'] as num?)?.toInt() ?? 0,
      budgetMin: (json['budgetMin'] as num?)?.toInt() ?? 0,
      budgetMax: (json['budgetMax'] as num?)?.toInt() ?? 0,
      timeline: json['timeline']?.toString() ?? '',
      scope: json['scope']?.toString() ?? '',
      status: json['status']?.toString() ?? 'OPEN',
      createdAt: json['createdAt']?.toString() ?? '',
      quoteCount: (json['quoteCount'] as num?)?.toInt() ?? 0,
      imageUrls: List<String>.from(json['imageUrls'] ?? []),
      customerName: json['customerName']?.toString(),
    );
  }
}

File: backend/src/main/java/com/interiorcolab/model/PropertyBrief.java
package com.interiorcolab.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "property_briefs")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class PropertyBrief {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id", nullable = false)
    private User customer;

    @Column(name = "property_type", nullable = false, length = 50)
    private String propertyType;

    @Column(nullable = false, length = 100)
    private String city;

    @Column(name = "size_sqft", nullable = false)
    private Integer sizeSqft;

    @Column(name = "budget_min", nullable = false, precision = 12, scale = 2)
    private BigDecimal budgetMin;

    @Column(name = "budget_max", nullable = false, precision = 12, scale = 2)
    private BigDecimal budgetMax;

    @Column(nullable = false, length = 100)
    private String timeline;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String scope;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private PropertyStatus status = PropertyStatus.OPEN;

    @ManyToMany
    @JoinTable(
        name = "property_targeted_firms",
        joinColumns = @JoinColumn(name = "property_id"),
        inverseJoinColumns = @JoinColumn(name = "firm_id")
    )
    private List<User> targetedFirms;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}
